### PR TITLE
add new dry dirt node

### PR DESCRIPTION
**Pull request by TumeniNodes**
This is the newest PR to add a new dry dirt node to Savannah
Everything tested, everything works.

The new texture - default_dry_dirt.png 
default_dry_dirt

<details><summary>Images (click to uncollapse)</summary>
<p>

screenshot_20190722_072042

mixed with old dirt with dry grass on old map
screenshot_20190722_072503

farming (you said you were neutral so I added it, because people will want it)
screenshot_20190722_073253

nice mixture of colours
screenshot_20190722_080308

<p>
</details>

### CODE REVIEW

**Comment by TumeniNodes** [comment]
A couple things I may need to attend too... in the images where the two are mixed on an old pre-existing map, notice there are no decorations on the newly generated dry_dirt_with_grass...

I need to look at why deco are not generating in this situation.

File: mods/default/mapgen.lua [hunk 1]
```diff
@@ -575,8 +576,8 @@ function default.register_ores()
 		},
 		-- Only where default:dirt is present as surface material
 		biomes = {"taiga", "snowy_grassland", "grassland", "coniferous_forest",
-			"deciduous_forest", "deciduous_forest_shore", "savanna", "savanna_shore",
-			"rainforest", "rainforest_swamp", "floatland_coniferous_forest"}
+	"deciduous_forest", "deciduous_forest_shore", "rainforest",
+	"rainforest_swamp", "floatland_coniferous_forest"}
```
Comment (paramat) [inline mods/default/mapgen.lua]: Needs proper indentation

File: mods/default/mapgen.lua [hunk 2]
```diff
@@ -1798,7 +1799,7 @@ local function register_dry_grass_decoration(offset, scale, length)
 	minetest.register_decoration({
 		name = "default:dry_grass_" .. length,
 		deco_type = "simple",
-		place_on = {"default:dirt_with_dry_grass"},
+		place_on = {"default:dirt_with_dry_grass", "default:dry_dirt_with_dry_grass"},
```
Comment (paramat) [inline mods/default/mapgen.lua]: Only place on 'dry dirt with ...'

**Comment by paramat** [comment]
PR looks near-perfect at first look.

File: mods/default/mapgen.lua (hunk 1 above)
Comment (TumeniNodes) [reply mods/default/mapgen.lua]: sorry, I misunderstood your comment here https://github.com/minetest/minetest_game/pull/2412#discussion_r304164406

fixed

File: mods/default/mapgen.lua (hunk 2 above)
Comment (TumeniNodes) [reply mods/default/mapgen.lua]: fixed

**Comment by sfan5** [comment]
The texture reminds me too much of clay, could be more "dusty" to match a Savannah.
:+1: otherwise

**Comment by TumeniNodes** [comment]
> The texture reminds me too much of clay, could be more "dusty" to match a Savannah.
> +1 otherwise

Thank you.
Hm, I'll have to figure out a way to work "dusty" into this 16x16 texture xD

File: mods/default/mapgen.lua (hunk 1 above)
Comment (paramat) [reply mods/default/mapgen.lua]: Oh i see, yes 1 relative indent.

**Comment by paramat** [comment]
Hmm, 'more dusty' might be difficult. Texture seems fine to me, make sure to keep a copy of this version.

**Comment by TumeniNodes** [comment]
> Hmm, 'more dusty' might be difficult. Texture seems fine to me, make sure to keep a copy of this version.

I always do ; )

**Comment by TumeniNodes** [comment]
I'm not sure if @sfan5 's comment is related to the colour or the texture itself.

**Comment by sfan5** [comment]
Mostly the texture, it was too smooth for my liking.
It's fine now.

**Comment by TumeniNodes** [comment]
Wonderful, thank you

**Comment by TumeniNodes** [comment]
with the new texture design, I simply set the colour back to the original.
That should be it as far as I can tell on my end.
I consider this ready as is.

I will follow up with a couple of farming changes @sfan5 mentioned in IRC earlier in another PR

**Comment by Napiophelios** [comment]
>I will follow up....in another PR

Ho, sh*t. You killing me, man.  :P

**Comment by paramat** [comment]
Latest texture is fine for me. But it needs to be converted to indexed colour.
Will test this now.

**Comment by TumeniNodes** [comment]
> Latest texture is fine for me. But it needs to be converted to indexed colour.
> Will test this now.

Sorry bout that, I could have sworn I did that, but may have forgot after I reset the colour. Done now

**Comment by TumeniNodes** [comment]
> Will test this now.

Testing intensifies. :P

**Comment by paramat** [comment]
Been very busy, testing now.

File: mods/default/functions.lua [hunk 3]
```diff
@@ -507,8 +506,6 @@ minetest.register_abm({
 		-- Most likely case first
```
Comment (paramat) [inline mods/default/functions.lua]: This comment now makes no sense, please remove

File: mods/default/mapgen.lua [hunk 4]
```diff
@@ -35,6 +35,7 @@ minetest.register_alias("mapgen_cobble", "default:cobble")
 minetest.register_alias("mapgen_stair_cobble", "stairs:stair_cobble")
 minetest.register_alias("mapgen_mossycobble", "default:mossycobble")
 minetest.register_alias("mapgen_stair_desert_stone", "stairs:stair_desert_stone")
+minetest.register_alias("mapgen_sandstonebrick", "default:sandstonebrick")
```
Comment (paramat) [inline mods/default/mapgen.lua]: ?

File: mods/farming/nodes.lua [hunk 5]
```diff
@@ -64,6 +80,32 @@ minetest.register_node("farming:soil_wet", {
 	}
 })
 
+minetest.register_node("farming:dry_soil", {
+	description = "Dry Soil",
+	tiles = {"default_dry_dirt.png^farming_soil.png", "default_dry_dirt.png"},
+	drop = "default:dirt",
```
Comment (paramat) [inline mods/farming/nodes.lua]: dry dirt

**Comment by paramat** [comment]
+1 ans will merge once line comments attended to.

File: mods/default/mapgen.lua (hunk 4 above)
Comment (TumeniNodes) [reply mods/default/mapgen.lua]: My goodness, absolutely no idea how that got in there? 0_o Very odd

**Comment by paramat** [comment]
:+1:

**Pull request merged**